bytepushers-common-js-package/src/main/javascript/com.bytepushers.base.app.js

```diff
@@ -122,6 +124,266 @@
+                if (this.getDate() + 1 === date.getDate()) {
+                    return true;
+                }
+            } else if (this.getMonth() + 1 === date.getMonth()) {
+                if (this.isLastDayInMonth() && date.getDate() === 1) {
+                    return true;
+                }
+            }
+        } else if (this.getFullYear() + 1 === date.getFullYear()) {
+            if (this.getMonth() === 11 && date.getMonth() === 0) {
+                if (this.getDate() === 31 && date.getDate() === 1) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    };
+
+    /**
+     * <p>Function that is used to determine a date is the day before another date.</p>
+     * @static
+     * @function
+     * @param {@link Date} The date to evaluate against this object.
+     * @return {<a href="http://www.w3schools.com/jsref/jsref_obj_boolean.asp">Boolean</a>} True if the date is the day before the original date; otherwise return false.
+     * @author <a href="mailto:pouncilt.developer@gmail.com">Tont&eacute; Pouncil</a>
+     */
+    Date.prototype.isDateEqualToYesterday = function (date) {
+        "use strict";
+        if (this.getFullYear() === date.getFullYear()) {
+            if (this.getMonth() === date.getMonth()) {
+                if (this.getDate() - 1 === date.getDate()) {
+                    return true;
+                }
+            } else if (this.getMonth() === date.getMonth() + 1) {
+                if (this.getDate() === 1 && date.isLastDayInMonth()) {
+                    return true;
+                }
+            }
+        } else if (this.getFullYear() - 1 === date.getFullYear()) {
+            if (this.getMonth() === 0 && date.getMonth() === 11) {
+                if (this.getDate() === 1 && date.getDate() === 31) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    };
+
+    /**
+     * <p>Tells you whether it is the last day in a month or not.</p>
+     * @private
+     *
+     * @returns {String} A new Date object with the specified time added.
+     * @author <a href="mailto:pouncilt.developer@gmail.com">Tont&eacute; Pouncil</a>
+     */
+    Date.prototype.isLastDayInMonth = function () {
+        "use strict";
+        var lastDayInMonth = this.getCurrentMonthTotalDays();
+        if (this.getDate() === lastDayInMonth) {
+            return true;
+        }
+        return false;
+    };
+
+    /**
+     * <p>Static function that tells you whether a date is the last day in a month or not.</p>
+     * @private
+     * @param {<a href="http://www.w3schools.com/jsref/jsref_obj_date.asp">Number</a>} time Represents the time you want to add to the date.
+     *
+     * @returns {String} A new Date object with the specified time added.
+     * @author <a href="mailto:pouncilt.developer@gmail.com">Tont&eacute; Pouncil</a>
+     */
+    Date.isLastDayInMonth = function (date) {
```

Comment:
Move all static method to the bottom of file and beside to add the @static annotation to the jsDoc for static methods.